flocker/ca/_script.py

```diff
@@ -16,10 +16,74 @@
 from ..common.script import (flocker_standard_options, ICommandLineScript,
                              FlockerScriptRunner)
 
-from ._ca import (CertificateAuthority, CertificateAlreadyExistsError,
-                  KeyAlreadyExistsError)
+from ._ca import (RootCredential, ControlCredential,
+                  CertificateAlreadyExistsError, KeyAlreadyExistsError,
+                  PathError)
 
 
+@flocker_standard_options
+class ControlCertificateOptions(Options):
+    """
+    Command line options for ``flocker-ca create-control-certificate``.
+    """
+
+    longdesc = """Create a new certificate for the control service.
+
+    Creates a certificate signed by a previously generated certificate
+    authority (see flocker-ca initialize command for more information).
+
+    The certificate will be stored in the specified output directory
+    (defaults to current working directory).
+    """
+
+    synopsis = "[OPTIONS]"
+
+    optParameters = [
+        ['inputpath', 'i', os.getcwd(),
+         'Path to directory containing root certificate.'],
+        ['outputpath', 'o', os.getcwd(),
+         'Path to directory to write control service certificate.'],
+    ]
+
+    def run(self):
+        """
+        Check if control service certificate already exist in current
+        directory. If it does, error out. Also check if root key and
+        certificate files (either default or as specified on the command line)
+        exist in the path and error out if they do not. If there are no path
+        errors, create a new control service certificate signed by the root
+        and write it out to the current directory.
+        """
+        self["inputpath"] = FilePath(self["inputpath"])
+        self["outputpath"] = FilePath(self["outputpath"])
+
+        d = Deferred()
```

Comment:
`Deferred` seems unnecessary for much of the code here. If external API requires it then return `succeed()` at the end, and normal blocking logic for the bulk of the code.